lib/bats-core/test_functions.bash

```diff
@@ -182,6 +182,107 @@ bats_separate_lines() { # <output-array> <input-var>
   fi
 }
 
+bats_pipe() { # [-N] [--] command0 [ \| command1 [ \| command2 [...]]]
+  # This will run each command given, piping them appropriately.
+  # Meant to be used in combination with `run` helper to allow piped commands
+  # to be used.
+  # Note that `\|` must be used, not `|`.
+  # By default, the exit code of this command will be the last failure in the
+  # chain of piped commands (similar to `set -o pipefail`).
+  # Supplying -N (e.g. -0) will instead always use the exit code of the command
+  # at that position in the chain.
+
+  local pipestatus_position=-1
```

Comment:
I suggest adding the possibility to use negative values like `--1` to always use the last status, `--2` next to last status, and so on
```suggestion
  local pipestatus_position=
```

Reply:
I suppose negative indexing (as used in linux's bash, python, etc) would be an interesting option. But the suggested syntax is quite peculiar. I don't think I've seen negative values used outside args to other parameters (e.g. `--num -5`, `-X=-100`, etc) or as a positional argument (e.g. `some_command -a -f -- -2 -50 -100`). Using negative values as optional arguments confuses the meaning of `-` and becomes more confusing when considering unix style arguments (where `--` is for long parameters).

WDYT about adding in a separate argument that allows for negative values? Something like `--status=N` (where `N` is a positive or negative value). The `-N` syntax would just be a shorthand (for positive values).

Reply:
I've gone ahead and added in `--returned-status` (which supports positive and negative values). I've also added in tests for this. Let me know what you think.